### PR TITLE
Improvement: Fix copy/paste when using the console

### PR DESCRIPTION
**Pull request by cocojoe**
Terminal code update to latest xterm major 4.x

This is a fix for #254 and I've added some additional notes as well in the thread.

I'm not a JS Engineer, so maybe it needs some tweaks.

Copy and paste in the console window should work now, you can also Select All. 
Managed to get the electron build running, works well 👍 

For fun, I tested the local dev server with latest versions of Edge, Chrome, Firefox. Safari in Windows and Mac.

This recording was with the Chrome Version Version 80.0.3987.163 (Official Build) (64-bit) in Catalina (10.15.4)
Apr-05-2020 15-41-32

### CODE REVIEW

**Comment by parachvte** [comment]
Too many changes in dependencies, I'll check them one by one later to make sure not break something.

**Comment by cocojoe** [comment]
@parachvte, in general, you probably want to do general dependencies bump every now and then as most new minors *should* be okay and reduces the number of issues flagged up in npm security audit.

The longer you leave it, the scarier it becomes to bump 😄

File: package-lock.json [hunk 1]
```diff
@@ -8556,7 +8556,8 @@
           "version": "2.1.1",
           "resolved": "https://registry.npmjs.org/ansi-regex/-/ansi-regex-2.1.1.tgz",
           "integrity": "sha1-w7M6te42DYbg5ijwRorn7yfWVN8=",
-          "dev": true
+          "dev": true,
+          "optional": true
```
Comment (cocojoe) [inline package-lock.json:8560]: You should be able to ignore these, auto added when using latest npm it seems.

**Comment by cocojoe** [comment]
@parachvte I've refactored to only include changes related to xterm 
Also added src/node_modules to .gitignore as it was slowing down the push operation.

**Comment by parachvte** [comment]
@cocojoe I remove the `fit()` call on `FitAddon`, cuz when it's called in `componentDidMount`, the terminal's parent node may not actually mounted and has available width and height. `xterm.js` complains about this in `Terminal.ts#L186` ("This API only accepts integer") because passed width and height is `NaN`.

One common solution here is to add a `setTimeout()` around the `fit()` call to wait for DOM rendering and then call `fit()`. But this I tried simply remove the `fit()` call and it works for me. Can you check my changes as well?

**Pull request merged**